tests/ui/feature-gates/feature-gate-return_type_notation.rs

```diff
@@ -0,0 +1,13 @@
+// edition: 2021
+
+#![feature(async_fn_in_trait)]
+//~^ WARN the feature `async_fn_in_trait` is incomplete
+
+trait Trait {
+    async fn m();
+}
+
+fn foo<T: Trait<m(..): Send>>() {}
+//~^ ERROR return type notation is experimental
```

Comment:
suggestion: also have a test for this under `#[cfg]`, to make extra sure we can remove it later if needed since this is non-RFC'd syntax.

Reply:
Done.